### PR TITLE
Check if Kata/ CNV operators are installed

### PR DESCRIPTION
**Pull request by ebattat**
Enhancement:
When running workloads, we are adding Kata/CNV operator verification. If the operator is not installed, an error will be raised. However, this will allow us to run the rest of the workloads even if one of the operators is not installed.

** The same fix should be added to clusterbuster

### CODE REVIEW

File: benchmark_runner/benchmark_operator/benchmark_operator_exceptions.py [hunk 1]
```diff
@@ -5,26 +5,45 @@ class BenchmarkOperatorError(Exception):
...
+class CNVNotInstalled(BenchmarkOperatorError):
     """
-    This class is error that ODF operator is not installed
+    This class raises an error that CNV operator is not installed
     """
-    def __init__(self):
-        self.message = "ODF is not installed, set 'ODF_PVC' to False"
-        super(ODFNonInstalled, self).__init__(self.message)
+    def __init__(self, workload):
+        self.message = f"{workload} is required CNV operator"
```
Comment (RobertKrawitz) [inline benchmark_runner/benchmark_operator/benchmark_operator_exceptions.py:13]: "requires CNV to be installed"

File: benchmark_runner/benchmark_operator/benchmark_operator_exceptions.py [hunk 2]
```diff
@@ -5,26 +5,45 @@ class BenchmarkOperatorError(Exception):
...
+        super(CNVNotInstalled, self).__init__(self.message)
+
+
+class KataNotInstalled(BenchmarkOperatorError):
+    """
+    This class raises an error that Kata operator is not installed
+    """
+    def __init__(self, workload):
+        self.message = f"{workload} is required Kata operator"
```
Comment (RobertKrawitz) [inline benchmark_runner/benchmark_operator/benchmark_operator_exceptions.py:22]: "requires Kata to be installed"

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py [hunk 3]
```diff
@@ -159,14 +159,14 @@ def make_undeploy_benchmark_controller_manager_if_exist(self, runner_path: str =
     @logger_time_stamp
     def login(self):
         """
-        This method login to the cluster
+        This method logins to the cluster
```
Comment (RobertKrawitz) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py:162]: "logs in"

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py [hunk 4]
```diff
@@ -383,7 +383,7 @@ def _create_run_artifacts(self, workload: str = '', database: str = '', labels:
 
     def __make_run_artifacts_tarfile(self, workload: str):
         """
-        This method tar.gz log path and return the tar.gz path
+        This method compresses tar.gz log path and returns the tar.gz path
```
Comment (RobertKrawitz) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py:386]: compresses the log file and returns the compressed path.

File: benchmark_runner/common/oc/oc.py [hunk 5]
```diff
@@ -431,7 +431,7 @@ def collect_events(self):
     @logger_time_stamp
     def login(self):
         """
-        This method login to the cluster
+        This method logins to the cluster
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:434]: logs in

File: benchmark_runner/common/oc/oc.py [hunk 6]
```diff
@@ -496,7 +496,7 @@ def describe_pod(self, pod_name: str, namespace: str = ''):
     @logger_time_stamp
     def get_pods(self):
         """
-        This method get pods
+        This method gets pods
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:499]: retrieves information on benchmark-runner pods in `oc get pod` format.

File: benchmark_runner/common/oc/oc.py [hunk 7]
```diff
@@ -659,7 +659,7 @@ def wait_for_ready(self, label: str, run_type: str = 'pod', workload: str = '',
     @typechecked
     def exec(self, command: str, pod_name: str, namespace: str = environment_variables.environment_variables_dict['namespace'], container: str = ''):
         """
-        oc exec a command and return the answer
+        This method execs a command and return the answer
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:662]: execs a command within a specified pod and optional container and returns the output

File: benchmark_runner/common/oc/oc.py [hunk 8]
```diff
@@ -722,7 +722,7 @@ def wait_for_pod_completed(self, label: str, workload: str = '', label_uuid: boo
                                namespace: str = environment_variables.environment_variables_dict['namespace'],
                                timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
         """
-        This method wait to pod to be completed
+        This method waits to pod to be completed
```
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:725]: waits for a pod to be completed

File: benchmark_runner/workloads/workloads_operations.py [hunk 9]
```diff
@@ -288,7 +288,7 @@ def _create_vm_run_artifacts(self, vm_name: str, start_stamp: str, end_stamp: st
 
     def __make_run_artifacts_tarfile(self, workload: str):
         """
-        This method tar.gz log path and return the tar.gz path
+        This method compresses tar.gz log path and returns the tar.gz path
```
Comment (RobertKrawitz) [inline benchmark_runner/workloads/workloads_operations.py:291]: compresses the log file and returns the compressed path

File: benchmark_runner/workloads/workloads_operations.py [hunk 10]
```diff
@@ -334,7 +334,7 @@ def upload_run_artifacts_to_s3(self):
 
     def __get_metadata(self, kind: str = None, status: str = None, result: dict = None) -> dict:
         """
-        This method return metadata kind and database argument are optional
+        This method returns metadata. The 'kind' arguments is optional.
```
Comment (RobertKrawitz) [inline benchmark_runner/workloads/workloads_operations.py:337]: returns metadata for a run, optionally filtered by runtime kind

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]
All comment cleanup, nothing substantive.

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py [hunk 11]
```diff
@@ -118,7 +118,7 @@ def update_node_selector(self, runner_path: str = environment_variables.environm
     @logger_time_stamp
     def make_deploy_benchmark_controller_manager(self, runner_path: str = environment_variables.environment_variables_dict['runner_path']):
         """
-        This method make deploy benchmark operator
+        This method makes deploy benchmark operator
```
Comment (RobertKrawitz) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py:121]: "This method deploys the benchmark operator"

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py [hunk 12]
```diff
@@ -135,7 +135,7 @@ def make_deploy_benchmark_controller_manager(self, runner_path: str = environmen
     @logger_time_stamp
     def make_undeploy_benchmark_controller_manager(self, runner_path: str = environment_variables.environment_variables_dict['runner_path']):
         """
-        This method make undeploy benchmark operator
+        This method makes undeploy benchmark operator
```
Comment (RobertKrawitz) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py:138]: "This method removes a benchmark-operator deployment"

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py [hunk 13]
```diff
@@ -148,7 +148,7 @@ def make_undeploy_benchmark_controller_manager(self, runner_path: str = environm
     @logger_time_stamp
     def make_undeploy_benchmark_controller_manager_if_exist(self, runner_path: str = environment_variables.environment_variables_dict['runner_path']):
         """
-        This method make undeploy benchmark controller manager if exist
+        This method makes undeploy benchmark controller manager if exist
```
Comment (RobertKrawitz) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads_operations.py:151]: "uninstalls/undeploys/removes benchmark controller manager, if it exists"

File: benchmark_runner/common/oc/oc.py (hunk 5 above)
Comment (RobertKrawitz) [inline benchmark_runner/common/oc/oc.py:434]: "logs in"

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]
A few more minor comment issues

**Comment by ebattat** [comment]
Thx, Please let me know if there are more issues

**Review by RobertKrawitz — APPROVED** [review]

**Pull request merged**